blockchain/difficulty_test.go

```diff
@@ -1150,3 +1150,260 @@ func TestMinDifficultyReduction(t *testing.T) {
+	g.AdvanceToHeight(expectedContinuityFailureHeight, 0)
+
+	// Ensure the chain continuity check fails at the expected failure height
+	// because there were no live tickets available to progress the chain
+	// after the next ticket maturity window.
+	tip = g.Tip()
+	tipHeight = tip.Header.Height
+	tipHash = tip.BlockHash()
+	liveTickets, err = g.chain.LiveTicketsAtNextMaturity(&tipHash)
+
+	if liveTickets != 0 {
+		teardown()
+		t.Fatalf("expected no live tickets at height #%d, got %d",
+			tipHeight, liveTickets)
+	}
+	if err == nil {
+		teardown()
+		t.Fatalf("expected a continuity error at height #%d", tipHeight)
+	}
+
+	teardown()
+
+	_, g, teardown = setup()
+
+	// Progress the chain to StakeValidationHeight buying TicketsPerBlock
+	// tickets each block.
+	g.AdvanceToHeight(stakeValidationHeight, ticketsPerBlock)
+
+	tip = g.Tip()
+	tipHeight = tip.Header.Height
+	tipHash = tip.BlockHash()
+	liveTickets, err = g.chain.LiveTicketsAtNextMaturity(&tipHash)
+
+	if liveTickets != expectedLiveTickets {
+		teardown()
+		t.Fatalf("expected %d live tickets at height #%d, got %d",
+			expectedLiveTickets, tipHeight, liveTickets)
+	}
+	if err != nil {
+		teardown()
+		t.Fatalf("unexpected continuity error at height #%d: %v", tipHeight, err)
+	}
+
+	ticketReUpHeight := expectedContinuityFailureHeight - (maturityWindow + 1)
+
+	// Progress the chain to the ticket reup height buying no additional tickets.
+	g.AdvanceToHeight(ticketReUpHeight-1, 0)
+
+	// Buy TicketPerBlock tickets at the ticket reup height.
+	g.AdvanceToHeight(ticketReUpHeight, ticketsPerBlock)
+
+	// Progress the chain to the expected chain continuity failure height
+	// buying no additional tickets.
+	g.AdvanceToHeight(expectedContinuityFailureHeight, 0)
+
+	// Ensure the chain continuity check passes at the prevously expected
+	// failure height because additional tickets were bought at the reup
+	// height.
+	tip = g.Tip()
+	tipHeight = tip.Header.Height
+	tipHash = tip.BlockHash()
+	liveTickets, err = g.chain.LiveTicketsAtNextMaturity(&tipHash)
+
+	if liveTickets != ticketsPerBlock {
+		teardown()
+		t.Fatalf("expected %d live tickets at height #%d, got %d",
+			ticketsPerBlock, tipHeight, liveTickets)
+	}
+	if err != nil {
+		teardown()
+		t.Fatalf("unexpected continuity error at height #%d: %v",
+			tipHeight, err)
+	}
+
+	// Progress the chain one more block without buying tickets.
+	g.AdvanceToHeight(expectedContinuityFailureHeight+1, 0)
+
+	// Ensure the chain continuity check fails because there would
+	// be no live tickets available to progress the chain.
+	tip = g.Tip()
+	tipHeight = tip.Header.Height
+	tipHash = tip.BlockHash()
+	liveTickets, err = g.chain.LiveTicketsAtNextMaturity(&tipHash)
+
+	if liveTickets != 0 {
+		teardown()
+		t.Fatalf("expected zero live tickets at height #%d, got %d",
+			tipHeight, liveTickets)
+	}
+	if err == nil {
+		teardown()
+		t.Fatalf("expected a continuity error at height #%d", tipHeight)
+	}
+
+	teardown()
+}
```

Comment:
There are a couple of code paths not being tested if you want to hit 100% coverage for `LiveTicketsAtNextMaturity`:

- https://github.com/decred/dcrd/blob/b1f2c12bbdcab81cd40b9b0f51f3a0c028a3a44e/blockchain/difficulty.go#L552
- https://github.com/decred/dcrd/blob/b1f2c12bbdcab81cd40b9b0f51f3a0c028a3a44e/blockchain/difficulty.go#L563
- https://github.com/decred/dcrd/blob/b1f2c12bbdcab81cd40b9b0f51f3a0c028a3a44e/blockchain/difficulty.go#L596-L599